scaladoc/src/dotty/tools/scaladoc/site/SidebarParser.scala

```diff
@@ -6,31 +6,76 @@ import com.fasterxml.jackson.databind.ObjectMapper;
+  )
+  case Page(title: Option[String], pagePath: String, hidden: Boolean)
 
 object Sidebar:
-  case class RawInput(var title: String, var page: String, var index: String, var subsection: JList[RawInput]):
-    def this() = this("", "", "", JList())
 
-    def setTitle(t: String) = this.title = t
-    def setPage(p: String) = this.page = p
-    def setIndex(i: String) = this.index = i
-    def setSubsection(s: JList[RawInput]) = this.subsection = s
+  type Child = Category | Page
+  case class RawRoot(var rootIndex: String, var pages: JList[RawInput]):
+    def this() = this("", JList())
 
-  type RawInnerTpe = JMap[String, JList[RawInput]]
-  private object RawTypeRef extends TypeReference[RawInnerTpe]
+    def setRootIndex(s: String) = rootIndex = s
+    def setPages(l: JList[RawInput]) = pages = l
 
-  private def toSidebar(r: RawInput): Sidebar = r match
-    case RawInput(title, page, index, subsection) if page.nonEmpty && index.isEmpty && subsection.isEmpty() || title == "Blog" =>
-      Sidebar.Page(Option.when(title.nonEmpty)(title), page)
-    case RawInput(title, page, index, subsection) if page.isEmpty && !subsection.isEmpty() =>
-      Sidebar.Category(Option.when(title.nonEmpty)(title), Option.when(index.nonEmpty)(index), subsection.asScala.map(toSidebar).toList)
+  case class RawInput(
+    @BeanProperty var title: String,
+    @BeanProperty var page: String,
+    @BeanProperty var index: String,
+    @BeanProperty var subsection: JList[RawInput],
+    @BeanProperty var directory: String,
+    @BooleanBeanProperty var hidden: Boolean
+  ):
+    def this() = this("", "", "", JList(), "", false)
 
-  def load(content: String): Seq[Sidebar] =
+  private object RootTypeRef extends TypeReference[RawRoot]
+
+  private def toSidebar(r: RawInput)(using CompilerContext): Sidebar.Child = r match
+    case RawInput(title, page, index, subsection, dir, hidden) if page.nonEmpty && index.isEmpty && subsection.isEmpty() =>
+      Sidebar.Page(Option.when(title.nonEmpty)(title), page, hidden)
+    case RawInput(title, page, index, subsection, dir, hidden) if page.isEmpty && (!subsection.isEmpty() || !index.isEmpty()) =>
+      Sidebar.Category(Option.when(title.nonEmpty)(title), Option.when(index.nonEmpty)(index), subsection.asScala.map(toSidebar).toList, Option.when(dir.nonEmpty)(dir))
+    case RawInput(title, page, index, subsection, dir, hidden) =>
+      report.error(s"Error parsing YAML configuration file.\n$schemaMessage")
+      Sidebar.Page(None, page, hidden)
+
+  private def schemaMessage: String =
+    s"""Static site YAML configuration file should comply to following model:
+      |case class Root(rootIndex: String, pages: List[Child])
+      |case class Page(title: Option[String], page: String) extends Child
+      |case class Subsection(
+      |  index: Option[String],
+      |  title: Option[String],
+      |  directory: Option[String],
+      |  subsection: List[Child]
+      |)
+      |""".stripMargin
```

Comment:
I am not sure we should show a Scala type hierarchy to document a YAML schema. Maybe we should put a link to a documentation web page if it’s not possible to simply describe the YAML format in a message? Also, this type hierarchy does not compile (the type `Child` is not defined).

Reply:
Ahh, I just didn't know how to define the format properly. I tried with JSON schema but it's too long. Maybe I'll try to just describe it in a less formal way.